gosrc/github.go

```diff
@@ -101,14 +102,24 @@ func getGitHubDir(client *http.Client, match map[string]string, savedEtag string
 		return nil, ErrNotModified
 	}
 
+	p, err := c.getBytes(expand("https://api.github.com/repos/{owner}/{repo}/contents{dir}?ref={tag}", match))
+	if err != nil {
+		return nil, err
+	}
+	p = bytes.TrimSpace(p)
+
+	// GitHb returns JSON array for directories
+	if len(p) == 0 || p[0] != '[' {
+		return nil, NotFoundError{Message: "Not a directory in GitHub"}
+	}
+
```

Comment:
I think the error you'd get would be something like:

```
error: json: cannot unmarshal object into Go value of type []struct { Type string; Name string; GitURL string "json:\"git_url\""; HTMLURL string "json:\"html_url\"" }
```

Which is [`json.UnmarshalTypeError`](https://godoc.org/encoding/json#UnmarshalTypeError). You can check if that's the error with something like:

```Go
if e, ok := err.(UnmarshalTypeError); ok && e.Value == "object" {
```

What do you think?